docs/adr/0005-remote-settings-client.md

```diff
@@ -0,0 +1,112 @@
+* There’s no requirement to manage buckets, groups and collections via this new library.
+
+### Requirements
+The requirements we do have are:
+* The library should allow fetching records from Mozilla’s Remote Settings servers. This includes support for attachments, and fetching [incremental changes](https://docs.kinto-storage.org/en/latest/api/1.x/filtering.html#polling-for-changes).
+* The library should not create threads or run any event loops - the mobile apps themselves are responsible for all threading requirements. While this might change in the future, considering this kind of change to our mobile applications is out of scope for this project.
+* We must use Necko for all networking on Android, must enforce all connections are via valid https hosts (although some test-only exceptions might be helpful for QA, such as allowing localhost connections to be http)
+* The library should be the only remote-settings library used in the browser. Specifically, this means that Nimbus must also be capable of using the library, and the work to move Nimbus to the library must be considered as part of the project.
+
+## Existing Libraries
+We have identified the following libraries which may be suitable for this project.
+### Remote-settings on desktop
+There is a version of the [remote settings client in desktop](https://searchfox.org/mozilla-central/source/services/settings/RemoteSettingsClient.jsm), written in Javascript.
+It is obviously very capable, but the roadblock to it being suitable for use by our mobile browsers is that it is written in Javascript, so there’s no reasonable path to have it made available to Swift and Kotlin.
+
+### Rust Remote Settings Client
+There is an existing [remote settings client on github](https://github.com/mozilla-services/remote-settings-client).
+This client is written in Rust and has evolved over a number of years.
+The most recent changes were made to support being used in [Merino](https://github.com/mozilla-services/merino), which is currently being re-written in Python - and once that rewrite is complete, there will be no known consumers of this library left.
+
+The main attributes of this library relevant to this discussion are:
+* It’s written in Rust, but has no FFI - ie, it’s currently only consumable by other Rust code.
+* It has recently been updated to use async rust, so requires an internal event loop.
+* It includes the capability to verify the signatures of the content.
+* It maintains a cache of remote resources as JSON files on disk.
+* It performs all network operations via standard Rust libraries.
+
+### The Nimbus-sdk Client
+The [nimbus-sdk is a component in the application-services repository](https://github.com/mozilla/application-services/tree/main/components/nimbus) written in Rust. It has client code which expects to talk to servers using the same format as the remote-settings client, but at time or writing, has only actually been used to talk to the experiments server.
```

Comment:
nit: s/or/of/

Reply:
3c356141e9177b963fbaf31aa07e9196b343cc27